x-pack/solutions/security/plugins/security_solution/public/attack_discovery/pages/settings_flyout/index.test.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+
+import { SettingsFlyout } from '.';
+import { useKibana } from '../../../common/lib/kibana';
+import { TestProviders } from '../../../common/mock';
+import { useSourcererDataView } from '../../../sourcerer/containers';
+import { ATTACK_DISCOVERY_SETTINGS } from './translations';
+
+jest.mock('../../../common/lib/kibana');
+jest.mock('../../../sourcerer/containers');
+jest.mock('react-router-dom', () => ({
+  matchPath: jest.fn(),
+  useLocation: jest.fn().mockReturnValue({
+    search: '',
+  }),
+  withRouter: jest.fn(),
+}));
+
+const defaultProps = {
+  end: undefined,
+  filters: undefined,
+  localStorageAttackDiscoveryMaxAlerts: undefined,
+  onClose: jest.fn(),
+  query: undefined,
+  setEnd: jest.fn(),
+  setFilters: jest.fn(),
+  setLocalStorageAttackDiscoveryMaxAlerts: jest.fn(),
+  setQuery: jest.fn(),
+  setStart: jest.fn(),
+  start: undefined,
+};
+
+const mockUseKibana = useKibana as jest.MockedFunction<typeof useKibana>;
+const mockUseSourcererDataView = useSourcererDataView as jest.MockedFunction<
+  typeof useSourcererDataView
+>;
+
+describe('SettingsFlyout', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    mockUseKibana.mockReturnValue({
```

Comment:
Same concern about this test possibly breaking if this PR merges before: https://github.com/elastic/kibana/pull/207878